plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -245,7 +259,7 @@ export class PersonState {
                     this.teamId,
                     this.timestamp
                 )
-            } else if (this.event.event === '$identify' && this.eventProperties['$anon_distinct_id']) {
+            } else if (this.event.event === '$identify' && '$anon_distinct_id' in this.eventProperties) {
```

Comment:
testing for every illegal id showed that even though we check for the empty string you couldn't get into this branch of the code if the anon distinct id was the empty string

Reply:
we would exit https://github.com/PostHog/posthog/pull/17325/files#diff-c51e19d65898f9cd1b4aaad32e5163e7b44507026ce49301507a2cbd6fe5ccdbR306